lib/grape/endpoint.rb

```diff
@@ -248,6 +248,8 @@ def run
 
         run_filters befores, :before
 
+        raise Grape::Exceptions::MethodNotAllowed, header.merge('Allow' => env[Grape::Env::GRAPE_METHOD_NOT_ALLOWED]) if env[Grape::Env::GRAPE_METHOD_NOT_ALLOWED]
```

Comment:
Maybe 1 will be aborted by rubocop, so I'd like to vote to 2 approach.
Btw, there are two reasons why I have pointed out.

1. By assiging variable, we can reduce number of characters of the line.
2. The assigned variable can be used in three places.